### PR TITLE
Preview state merge

### PR DESCRIPTION
**Pull request by JavaLeg**
Review pls:

Refactored UI:
State (Grid) -- Tile(GridCell)
EditorModel (Serialized grid)
Object Sub-types removed
EditorScreen Stages + Custom viewports (Toolbar, Preview, Editor)
Minor changes to the other screens

### CODE REVIEW

File: LibGDXGradle/desktop/src/Interface/Screens/EditorScreen.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+import Interface.Stages.Toolbar;
+import Interface.Viewports.CameraTestMain;
+import Interface.Viewports.PreviewViewport;
+import Interface.Viewports.ToolbarViewport;
+import State.State;
+
+
+public class EditorScreen implements Screen {
+	
+	protected ArrayList<Stage> UI;
```
Comment (JavaLeg) [inline LibGDXGradle/desktop/src/Interface/Screens/EditorScreen.java]: Someone can't indent. 
`call(police);`

File: LibGDXGradle/desktop/src/Interface/Screens/LibraryScreen.java [hunk 2]
```diff
@@ -54,22 +58,30 @@ public void show() {
         skin = new Skin(Gdx.files.internal("uiskin.json"), atlas);
```
Comment (JavaLeg) [inline LibGDXGradle/desktop/src/Interface/Screens/LibraryScreen.java]: Again indent

File: LibGDXGradle/desktop/src/Interface/Screens/LibraryScreen.java [hunk 3]
```diff
@@ -35,15 +40,14 @@
     private OrthographicCamera camera;
```
Comment (JavaLeg) [inline LibGDXGradle/desktop/src/Interface/Screens/LibraryScreen.java]: Indent again

File: LibGDXGradle/desktop/src/Interface/Screens/LibraryScreen.java (hunk 2 above)
Comment (JavaLeg) [inline LibGDXGradle/desktop/src/Interface/Screens/LibraryScreen.java]: Indent again

File: LibGDXGradle/desktop/src/Interface/Screens/MainMenuScreen.java [hunk 4]
```diff
@@ -64,9 +63,6 @@ public void show() {
          * Add buttons to table
          * Add table to stage
          */
-        
-        
-
         //Create buttons
         TextButton playButton = new TextButton("Editor", skin);
```
Comment (JavaLeg) [inline LibGDXGradle/desktop/src/Interface/Screens/MainMenuScreen.java]: !!

**Comment by SamWiseKimchi** [comment]
holy shit 228 files.

**Pull request merged**